src/main/java/de/craften/ui/swingmaterial/MaterialComboBox.java

```diff
@@ -38,19 +98,23 @@ public void paint(Graphics g) {
         g2.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
         g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
 
-        g.setColor(Color.WHITE);
+        g.setColor(getBackground());
         g.fillRect(0, 0, getWidth(), getHeight());
 
         g.setFont(Roboto.REGULAR.deriveFont(16f));
-        g2.setColor(Color.BLACK);
+        g.setColor(getSelectedItem() == null ? new Color(getForeground().getRGB() & 0x00FFFFFF | HINT_OPACITY_MASK, true):getForeground());
```

Comment:
I've decided to move it into a method in Utils, so it can be properly commented and shared between components.